File: Mathematics/juggler sequence/javascript/juggler_seq.js
(function(){
"use strict"

const juggle = function(n) {
	var prev,next
	var result = [];

	prev = n;

	//the first term is n
	result.push(prev);
	//continue until we reach the number 1
	while (prev != 1) {
		next = 0;

		if (prev%2 === 0) { //if even
			//next is the square root of the prev
			next = parseInt(Math.sqrt(prev));
		} else { //if odd
			//next is the prev to the power of 3/2
			next = parseInt(Math.pow(Math.sqrt(prev),3));
		}

		result.push(next);
		prev = next;
	}

	return result;
}

module.exports = juggle;

})();  

File: Mathematics/juggler sequence/C/juggler_seq.c
// C implementation of Juggler Sequence 
#include<stdio.h> 
#include<math.h> 

// This function prints the juggler Sequence 
void printJuggler(int n) 
{ 
	int a = n; 

	// print the first term 
	printf("%d ", a); 

	// calculate terms until last term is not 1 
	while (a != 1) 
	{ 
		int b = 0; 

		// Check if previous term is even or odd 
		if (a%2 == 0) 

			// calculate next term 
			b = floor(sqrt(a)); 

		else

			// for odd previous term calculate 
			// next term 
			b = floor(sqrt(a)*sqrt(a)*sqrt(a)); 

		printf("%d ", b); 
		a = b; 
	} 
} 

//driver program to test above function 
int main() 
{  
    int n;
    printf("Enter any Integer\n");
    scanf("%d",&n);
	printJuggler(n); 
	return 0; 
} 
